src/chaos/controller.go

```diff
@@ -0,0 +1,210 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+)
+
+type ChaosController struct {
+	DynamicClient dynamic.Interface
+	Namespace     string
+}
+
+// NewController creates a new instance of ChaosController
+func NewController(namespace string) *ChaosController {
+	config, err := client.GetConfig()
+	if err != nil {
+		log.Fatal(err.Error())
+		return nil
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Failed to create Kubernetes client: %v", err)
+	}
+
+	return &ChaosController{dynamicClient, namespace}
+}
+
+func (cc *ChaosController) listNodes(namespace string) ([]string, error) {
+	podGVR := schema.GroupVersionResource{
+		Group:    "", // Core API group
+		Version:  "v1",
+		Resource: "pods",
+	}
+
+	podList, err := cc.DynamicClient.Resource(podGVR).Namespace(namespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	alivePods := make([]string, 0)
+	for _, pod := range podList.Items {
+		alivePods = append(alivePods, pod.GetName())
+	}
+
+	return alivePods, nil
+}
+
+func (cc *ChaosController) deleteNode(namespace string, podName string) error {
+	podGVR := schema.GroupVersionResource{
+		Group:    "", // Core API group
+		Version:  "v1",
+		Resource: "pods",
+	}
+
+	return cc.DynamicClient.Resource(podGVR).Namespace(namespace).Delete(context.TODO(), podName, v1.DeleteOptions{})
+}
+
+// configConfigPath    - path to k8s cluster config
+// killNodes 	- to kill or not to kill affected nodes after each run
+func RunChaosExperiment(namespace string, chaosConfigPath string) error {
+	cc := NewController(namespace)
+
+	// Work with experiment yaml file
+	experimentFile, err := os.ReadFile(chaosConfigPath)
+	if err != nil {
+		log.Fatalf("Failed to read manifest file: %v", err)
+		return err
+	}
+	reader := bytes.NewReader(experimentFile)
+	decoder := yaml.NewYAMLOrJSONDecoder(reader, 1000)
+	parsedYaml := &unstructured.Unstructured{}
+
+	if err := decoder.Decode(parsedYaml); err != nil {
+		log.Fatalf("Failed to decode YAML: %v", err)
+		return err
+	}
+
+	expType, _, err := unstructured.NestedString(parsedYaml.Object, "kind")
+	if err != nil {
+		return err
+	}
+	gvr := schema.GroupVersionResource{
+		Group:    "chaos-mesh.org",         // ChaosMesh group
+		Version:  "v1alpha1",               // Version of the ChaosMesh resource
+		Resource: strings.ToLower(expType), // Resource is "podchaos"
+	}
+
+	// Apply the resource (equivalent to `kubectl apply -f`)
+	_, err = cc.DynamicClient.Resource(gvr).Namespace(cc.Namespace).Create(context.TODO(), parsedYaml, v1.CreateOptions{})
+	if err != nil {
+		log.Fatalf("Failed to apply ChaosMesh resource: %v", err)
+	}
+
+	return err
+}
+
+// Deleting the experiment from chaosmesh list
+// to avoid collisions, when the same experiment is started
+func ClearChaosCache(namespace string) error {
```

Comment:
Maybe it's better to choose the name `StopAllChaosExperiments`?

Reply:
Chaos experiment stops by itself when duration time has passed. After experiment finished, in k8s scope lays some metadata about passed experiment. This metadata won't allow to run experiment with name ```master-failure``` twice in a row, for example. Thus, we need to clear this sort of Cache or Metadata (don't know which name is better) to run the same chaos test twice